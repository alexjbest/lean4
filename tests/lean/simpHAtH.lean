axiom q : Nat → Prop
axiom p : Nat → Prop
axiom q_eq_p : q = p

example (h : ¬ q 0) : ¬ p 0 := by
  trace_state
  /-
     h : ¬ q 0
     ⊢ ¬ p 0
  -/
  simp [h] at h
  /-
     h : ¬ q 0
     ⊢ ¬ p 0
  -/
  trace_state
  rw [← q_eq_p]
  assumption
