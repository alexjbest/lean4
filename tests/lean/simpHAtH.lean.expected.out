h : ¬q 0
⊢ ¬p 0
h : ¬q 0
⊢ ¬p 0
